Jenkinsfile

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env groovy
+
+pipeline {
+    agent {
+        label "java"
+    }
+    tools {
+        maven 'mvn'
+        jdk 'jdk8'
+    }
+    options {
+        timestamps()
+    }
+
+    stages {
+        stage('Checkout') {
+            steps {
+                deleteDir()
+                checkout scm
```

Comment:
Declarative Pipeline automatically does a `checkout scm` unless you explicitly override this in `options`. So IIUC you are doing two checkouts per build.